app/queries/servers/team.test.ts

```diff
@@ -0,0 +1,1023 @@
+            expect(result.length).toBeGreaterThan(0);
+        });
+
+        it('should add channel to start of existing team history', async () => {
+            const channelId = 'channel1';
+            const existingChannelId = 'channel2';
+
+            await operator.handleMyChannel({
+                myChannels: [
+                    {
+                        id: channelId,
+                        roles: '',
+                        channel_id: '',
+                        user_id: '',
+                        last_viewed_at: 0,
+                        msg_count: 0,
+                        mention_count: 0,
+                        notify_props: {},
+                        last_update_at: 0,
+                    },
+                    {
+                        id: existingChannelId,
+                        roles: '',
+                        channel_id: '',
+                        user_id: '',
+                        last_viewed_at: 0,
+                        msg_count: 0,
+                        mention_count: 0,
+                        notify_props: {},
+                        last_update_at: 0,
+                    },
+                ],
+                channels: [{
+                    id: channelId,
+                    team_id: teamId,
+                    create_at: 0,
+                    update_at: 0,
+                    delete_at: 0,
+                    type: 'O',
+                    display_name: '',
+                    name: '',
+                    header: '',
+                    purpose: '',
+                    last_post_at: 0,
+                    total_msg_count: 0,
+                    extra_update_at: 0,
+                    creator_id: '',
+                    scheme_id: null,
+                    group_constrained: null,
+                    shared: false,
+                }, {
+                    id: existingChannelId,
+                    team_id: teamId,
+                    create_at: 0,
+                    update_at: 0,
+                    delete_at: 0,
+                    type: 'O',
+                    display_name: '',
+                    name: '',
+                    header: '',
+                    purpose: '',
+                    last_post_at: 0,
+                    total_msg_count: 0,
+                    extra_update_at: 0,
+                    creator_id: '',
+                    scheme_id: null,
+                    group_constrained: null,
+                    shared: false,
+                }],
+                prepareRecordsOnly: false,
+            });
+
+            await operator.handleTeamChannelHistory({
+                teamChannelHistories: [{
+                    id: teamId,
+                    channel_ids: [existingChannelId],
+                }],
+                prepareRecordsOnly: false,
+            });
+
+            const result = await addChannelToTeamHistory(operator, teamId, channelId);
+            expect(result.length).toBeGreaterThan(0);
```

Comment:
Here it would make sensehere it would make sense to check that the channelId is the first item on the result, but I get this is only for coverage